mongo/filterstore.go

```diff
@@ -422,19 +422,24 @@ func createUpdateFilterBlueprint(filter *models.Filter, currentTime time.Time) b
 func createUpdateFilterOutput(filter *models.Filter, currentTime time.Time) bson.M {
 
 	var downloads models.Downloads
-
+	state := "created"
 	var update bson.M
+	if filter.Downloads != nil {
+		if filter.Downloads.XLS.URL != "" {
```

Comment:
is it possible for the Download.XLS to be nil? if so put this inside an if check

Reply:
I think it is good practice to check XLS is not nil.

Reply:
The Downloads type within a filter-output is a pointer, so it can be nil. Both the CSV and XLSX types within the Downloads are not pointers, so they can not be nil. Checking for nil will trigger a compile error.